### DISCUSSION TITLE
Does pocketbase support sqlite auto-increment ?

### DISCUSSION THREAD

**Discussion by Jeyanthinath** [pocketbase/pocketbase#3275] [Q&A]
For my use-case, I need to insert the records into the table with some auto-increment field (https://www.sqlite.org/autoinc.html). 

The reason I need auto-incremental field is, my URL slug won't feel like some random UUID.

I don't want to insert the number in an incremental fashion from the code as it might not satisfy atomicity. 

currently: using the `id` field auto created by pb, I can only be able to achieve. these kind of URLs 

`http://localhost:3000/user/profile/gi7o2a3cxay8c4t`

but I need something like this. 

`http://localhost:3000/user/profile/1`
`http://localhost:3000/user/profile/2`
`http://localhost:3000/user/profile/3`

any ways of achieving this with the current setup?

**Comment by ganigeorgiev** [comment]
No this is not supported at the moment (_in the future there are plans for a more "native" SQLite dev experience and to allow more control over the system fields, but for now remains a low priority_).

If the only reason you want an auto-increment field is to make the url path more easy readable, then I'd suggest to reconsider it, especially for auth related records. From security standpoint, using a random string as resource identifier is often considered better as it prevents enumeration attacks.
